performance_manager/lib/rt_trip_updates.py

```diff
@@ -155,18 +160,115 @@ def get_and_unwrap_tu_dataframe(
     events["is_moving"] = False
     events["pk_id"] = None
 
+    return events
+
+
+def join_gtfs_static(
+    trip_updates: pandas.DataFrame, db_manager: DatabaseManager
+) -> pandas.DataFrame:
+    """
+    join trip update dataframe to gtfs static records
+
+    adds "fk_static_timestamp" and "parent_station" columns to dataframe
+    """
+    # get unique "start_date" values from trip update dataframe
+    # with associated minimum "timestamp_start"
+    date_groups = trip_updates.groupby(by="start_date")["timestamp_start"].min()
+
+    # pylint: disable=duplicate-code
+    # TODO(ryan): the following code is duplicated in rt_vehicle_positions.py
+
+    # dictionary used to match minimum "timestamp_start" values to
+    # "timestamp" from StaticFeedInfo table, to be used as foreign key
+    timestamp_lookup = {}
+    for (date, min_timestamp) in date_groups.iteritems():
+        date = int(date)
+        min_timestamp = int(min_timestamp)
```

Comment:
why the type conversion? i'm guessing they are stored as something else and `sa` needs them to be ints?

Reply:
These values are stored as numpy `Int64` dtypes in the dataframe, which is a dtype that sqlalchemy does not seem to be a fan of. Sqlalchemy throws an error in the select statement below without the conversion to the Python `int` dtype